CMakeLists.txt

```
@@ -0,0 +1,260 @@
+check_include_file(inttypes.h HAVE_INTTYPES_H)
+check_include_file(memory.h HAVE_MEMORY_H)
+check_include_file(stdlib.h HAVE_STDINT_H)
+check_include_file(stdint.h HAVE_STDLIB_H)
+check_include_file(strings.h HAVE_STRINGS_H)
+check_include_file(string.h HAVE_STRING_H)
+check_include_file(sys/stat.h HAVE_SYS_STAT_H)
+check_include_file(sys/types.h HAVE_SYS_TYPES_H)
+check_include_file(unistd.h HAVE_UNISTD_H)
+
+# ##############################################################################
+# library
+# ##############################################################################
+
+set(PACKAGE_STRING "${PROJECT_NAME} ${PROJECT_VERSION}")
+
+set(PACKAGE_VERSION "${PROJECT_VERSION}")
+set(VERSION "${PROJECT_VERSION}")
+configure_file(${CMAKE_CURRENT_LIST_DIR}/mbus/config.h.in ${CMAKE_CURRENT_LIST_DIR}/config.h)
+
+add_library(
+  ${PROJECT_NAME}
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-protocol.c
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-protocol.h
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-tcp.c
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-tcp.h
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus.c
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus.h
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-protocol-aux.c
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-protocol-aux.h
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-serial.c
+  ${CMAKE_CURRENT_LIST_DIR}/mbus/mbus-serial.h)
+target_include_directories(
+  ${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}> $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+                         $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+target_link_libraries(${PROJECT_NAME} PRIVATE m)
+target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wextra -Wno-pedantic)
+
+if(CLANG_TIDY_EXE)
+  set_target_properties(${PROJECT_NAME} PROPERTIES CXX_CLANG_TIDY "${DO_CLANG_TIDY}")
+endif()
+
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+
+# ##############################################################################
+# examples
+# ##############################################################################
+
+if(LIBMBUS_BUILD_EXAMPLES)
+  message(STATUS "building examples")
+  add_subdirectory(bin)
+endif()
+
+# ##############################################################################
+# tests
+# ##############################################################################
+
+if(LIBMBUS_BUILD_TESTS)
+  message(STATUS "building tests")
+  enable_testing()
+  add_subdirectory(test)
+endif()
+
+# ##############################################################################
+# install
+# ##############################################################################
+
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+
+set(LIBMBUS_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+install(
+  TARGETS ${PROJECT_NAME}
+  EXPORT ${PROJECT_NAME}Targets
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT lib
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT lib)
+
```

Windows provides runtime dll's
```suggestion
install(
  TARGETS ${PROJECT_NAME}
  EXPORT ${PROJECT_NAME}Targets
  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT lib
  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT lib
  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT lib)

```